markdowns/20_Using_Haystack_with_REST_API.md

```diff
@@ -0,0 +1,312 @@
+
+      ```yaml
+      environment:
+        ...
+        - PIPELINE_YAML_PATH=/opt/pipelines/document-search.haystack-pipeline.yml
+        ...
+      ```
+   4. Define the pipeline nodes in the `components` section of the file. A document search pipeline requires a DocumentStore and a Retriever. Our pipeline will use `ElasticsearchDocumentStore` and `EmbeddingRetriever`:
+
+      ```yaml
+      components:
+        - name: DocumentStore #How you want to call this node here
+          type: ElasticsearchDocumentStore #This is the Haystack node class
+          params: #The node parameters
+            embedding_dim: 384 #This parameter is required for the embedding_model
+        - name: Retriever
+          type: EmbeddingRetriever
+          params:
+            document_store: DocumentStore
+            top_k: 10
+            embedding_model: sentence-transformers/all-MiniLM-L6-v2
+      ```
+
+   5. Create a query pipeline in the `pipelines` section. Here, `name` refers to the name of the pipeline, and `nodes` defines the order of the nodes in the pipeline: 
+
+      ```yaml
+      pipelines:
+        - name: query 
+          nodes:
+            - name: Retriever
+              inputs: [Query]
+      ```
+
+2. In the same YAML file, create an indexing pipeline. This pipeline will index your documents to Elasticsearch through `rest_api`. 
+   1. Define `FileTypeClassifier`, `TextConverter`, and `PreProcessor` nodes for the pipeline:
+
+      ```yaml
+      components:
+        ...
+        - name: FileTypeClassifier
+          type: FileTypeClassifier
+        - name: TextFileConverter
+          type: TextConverter
+        - name: Preprocessor
+          type: PreProcessor
+          params: #These parameters define how you want to split your documents
+            split_by: word
+            split_length: 250
+            split_overlap: 30 
+            split_respect_sentence_boundary: True 
+      ```
+
+   2. In the `pipelines` section of the YAML file, create a new pipeline called `indexing`. In this pipeline, indicate how the nodes you just defined are connected to each other, Retriever, and DocumentStore. This indexing pipeline supports *.TXT* files and preprocesses them before loading to Elasticsearch.
+
+      ```yaml
+      pipelines:
+        ...
+        - name: indexing
+          nodes:
+            - name: FileTypeClassifier
+              inputs: [File]
+            - name: TextFileConverter
+              inputs: [FileTypeClassifier.output_1]
+            - name: Preprocessor
+              inputs: [TextFileConverter]
+            - name: Retriever
+              inputs: [Preprocessor]
+            - name: DocumentStore
+              inputs: [Retriever]
+      ```
+
+3. After creating query and indexing pipelines, add `version: ignore` to the top of the file. Now, the pipeline YAML is ready.
```

Comment:
Again, maybe too much detail and nitpicking but in paranthesis maybe a note on why we `ignore`?